### ISSUE TITLE
Patch: to correctly rename netdisco-daemon.log files when rotating

### ISSUE THREAD

**Issue by jrbinks** [netdisco/netdisco#740]
In relation to #731, I manually modified netdisco-backend to increase number of logfiles stored (I'm trying to figure out another issue), around line 165.  Unfortunately this doesn't work with the code as constituted, partly because of the glob I think -- it was a few weeks ago and I've forgotten the details.  It was something about the glob catches more than just netdisco-backend.log files, as there are netdisco-web.log files in there too, and then the renaming goes wrong.  Oh and it's also looking for a single-digit suffix.

I modified it thusly, which now works for me, although it doesn't seem very elegant.  Basically, it ensures that the list of logfiles contains only the netdisco-backend.log ones, extracts a list of the indexes of those files, then does a numerical reverse sort on them so that they rename in the correct order.  Note I've also set the number to keep to 20 in this code, from the default of 7.

```
--- /home/netdisco/perl5/bin/netdisco-backend   2020-07-08 14:06:57.987060000 +0100
+++ netdisco-backend.new        2020-07-13 09:53:25.564224000 +0100
@@ -158,11 +158,15 @@
   return unless (-f $log_file) and
     ((-s $log_file) > (10 * 1024768));

-  my @files = glob file($log_dir, '*');
-  foreach my $f (reverse sort @files) {
-      next unless $f =~ m/$log_file\.(\d)$/;
-      my $pos = $1;
-      unlink $f if $pos == 7;
+  my @allfiles = glob file($log_dir, '*');
+  my @files = grep { /$log_file\.\d+/ } @allfiles;
+  # extract just the indexes:
+  s/.+\.(\d+)/$1/ for @files;
+  foreach my $findex (sort { $b <=> $a } @files) {
+      my $f = "$log_file\.$findex";
+      my $pos = $findex;
+      next unless $f =~ m/$log_file\.(\d+)$/;
+      unlink $f if $pos == 20;
       my $next = $pos + 1;
       (my $newf = $f) =~ s/\.$pos$/.$next/;
       rename $f, $newf;
```

I would still like the number to keep to be controllable from the config (or a new 'backend' config), per my last comment on #731 :) .

App::Netdisco | 2.45.6

**Comment by inphobia** [comment]
this might be related to #509 and/or #472 
(where certain commandline arguments aren't saved between HUP restarts). i recall seeing related issues with log rotation, but never had time nor priorities to delve deeper into this.

**Comment by jrbinks** [comment]
No-one interested in this or #731 ? :) 

Not convinced it has anything to do with the bugs @inphobia mentioned.

**Comment by ollyg** [comment]
Hi @jrbinks 

First just to note thanks for being patient on this. Also yes the daemon scripts do "know" the config file but they do not have the intelligence to parse it - they are just making a sanity check that the file is in place before starting the app.

However I realised we have a simpler option:
[master 72934d39] #740 --logfiles=8 and --logsize=10 (MB) options to netdisco-backend and netdisco-web

This small patch allows you to pass `--logfiles=8` and `--logsize=10` (those are the default values) to the two daemon scripts (not the -fg versions of course). The logsize is in megabytes.

Hope this helps, it will be in the next release.

**Comment by jrbinks** [comment]
OK, thanks, but that's really a fix for my original #731  !

This ticket #740 is about bugs in the rotation that means it doesn't work for higher numbers, with an inelegant patch at the top that fixes it.

Sorry, thanks, PITA, Jethro :)

**Comment by ollyg** [comment]
fixed, I hope, in
[master b7ac3a3b] #740 more robust log rotation

**Comment by jrbinks** [comment]
I have updated to the version with this fix.

Now:
